extensions/command-bar/command-bar.ui.runtime.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Slot, SlotRegistry } from '@teambit/harmony';
+import Mousetrap from 'mousetrap';
+
+import UIAspect, { UIRuntime, UiUI } from '@teambit/ui';
+import { CommandBar } from './ui/command-bar';
+import { CommandSearcher } from './ui/command-searcher';
+import { CommandBarAspect } from './command-bar.aspect';
+import { commandBarCommands } from './command-bar.commands';
+import { SearchProvider, Keybinding, CommandHandler, CommandId } from './types';
+
+const RESULT_LIMIT = 5;
+type SearcherSlot = SlotRegistry<SearchProvider>;
+type CommandSlot = SlotRegistry<CommandEntry[]>;
+
+export type CommandEntry = {
+  id: CommandId;
+  handler: CommandHandler;
+  keybinding?: Keybinding;
+  displayName: string;
+};
+
+/** Quick launch actions. Use the `addSearcher` slot to extend the available actions */
+export class CommandBarUI {
+  private mousetrap = new Mousetrap();
+  private commandRegistry = new Map<CommandId, CommandEntry>();
+  private commandSearcher = new CommandSearcher([]);
+
+  /** Opens the command bar */
+  open = () => {
```

Comment:
why do you use arrow functions?

Reply:
using arrow functions to bind these method to the class